lib/Support/PrettyPrinter.cpp

```diff
@@ -114,8 +115,13 @@ void PrettyPrinter::add(Token t) {
         if (scanStack.empty())
           return print({t, 0});
         addScanToken(-1);
+      })
+      .Case([&](CallbackToken *c) {
+        tokens.push_back({t, 0});
```

Comment:
I don't know that this is right, looks like this interacts badly with other code paths re:scanStack being empty (at least).

Reply:
Updated the logic, push to `tokens` only if the `scanStack` is empty.